### PR TITLE
Model Incident to GUI

### LINKED ISSUE #42: Add GUI support for Incidents model

### PR DESCRIPTION
**Pull request by Yoshi275**
Resolve #42 

**Outdated Draft Comment:**
Not working yet. To ignore all issues with Incident loading to GUI,
go to `AddressBook#resetData(ReadOnlyAddressBook new Data)` and comment out `setIncidents` method.

Most GUI syncing files seem to have been created. Further abstraction may be needed for `Incident` class, although first priority is to load the incidents into the GUI. 

image
Screenshot of problematic display thus far

### CODE REVIEW

**Comment by Yoshi275** [comment]
**Update:**

Incidents model has been successfully integrated into GUI.

Tasks moving forward:
- Strategise and discuss Incident model creation (there are presently two constructors)
- Discuss changes to Incident Model - Person operator is contained within the Incident model, while callerNumber remains a String. What further information should we take from Person operator, for display in the GUI?

image
Updated Screenshot

File: src/main/java/seedu/address/model/incident/CallerNumber.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public class CallerNumber {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Caller numbers should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
```
Comment (madanalogy) [inline src/main/java/seedu/address/model/incident/CallerNumber.java]: Have you considered modelling caller number constraints after Person/Phone model instead?

File: src/main/java/seedu/address/model/incident/Incident.java [hunk 2]
```diff
@@ -13,14 +23,14 @@
 
     //is autofilled
     private Person operator;
-    private LocalDateTime dateTime;
+    private IncidentDateTime incidentDateTime;
     private Vehicle car;
     private IncidentId id;
```
Comment (madanalogy) [inline src/main/java/seedu/address/model/incident/Incident.java]: Could also consider making these final :)

File: src/main/java/seedu/address/model/incident/IncidentDateTime.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents an Incident's IncidentDateTime in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIncidentDateTime(String)}
+ */
+public class IncidentDateTime {
```
Comment (madanalogy) [inline src/main/java/seedu/address/model/incident/IncidentDateTime.java]: Maybe for future increments we can consider having this extend from a common DateTime object?

File: src/main/java/seedu/address/model/incident/CallerNumber.java (hunk 1 above)
Comment (Yoshi275) [reply src/main/java/seedu/address/model/incident/CallerNumber.java]: Good idea! Changed that in my last commit.

File: src/main/java/seedu/address/model/incident/Incident.java (hunk 2 above)
Comment (Yoshi275) [reply src/main/java/seedu/address/model/incident/Incident.java]: Oops missed out on that. Done!

**Pull request merged**